packages/components/addon/components/hds/pagination/size-selector/index.js

```diff
@@ -17,12 +18,7 @@ export default class HdsPaginationSizeSelectorComponent extends Component {
    * @description Set the page sizes users can select from.
    */
   get sizes() {
-    let { sizes } = this.args;
-
-    assert(
-      '@sizes for "Pagination::SizeSelector" must be defined',
-      sizes !== undefined
-    );
+    let { sizes = DEFAULT_PAGE_SIZES } = this.args;
```

Comment:
I would suggest to declare the default sizes not here, but in the `Pagination::Numbered` component. The advantage in this case is that we can make also the `itemsPerPage` optional, inferring it as first item of the `pageSizes` argument. This will simplify a lot the API of the component (and the documentation itself, now that we don't need to add both values for every instance).

(And the fact that we don't have a default when used as standalone element, is not a problem for me: if the consumer need to build their own pagination using the base elements, defining an array with default sizes is the last of their problems).

Reply:
Pushed updates